akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1076,6 +1076,56 @@ trait FlowOps[+Out, +Mat] {
     via(new Timers.IdleInject[Out, U](maxIdle, injectedElem))
 
   /**
+   * Sends elements downstream with speed limited to `elements/per`. In other words, this stage set the maximum rate
+   * for emitting messages. This combinator works for streams where all elements have the same cost or length.
+   *
+   * If there were no messages for sometime, throttle can emit `maximumBurst` number of elements at once, without
+   * waiting.
+   *
+   * Parameter `mode` manages behaviour when upstream is faster than throttle rate:
+   *  - [[akka.stream.ThrottleMode.Shaping]] makes pauses before emitting messages to meet throttle rate
+   *  - [[akka.stream.ThrottleMode.Enforcing]] fails with exception when upstream is faster than throttle rate
+   *
+   * '''Emits when''' upstream emits an element and configured time per each element elapsed
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def throttle(elements: Int, per: FiniteDuration, maximumBurst: Int,
```

Comment:
we can explicitly say `tokenCapacity` and then explain that tokens can be "spared" for later use up to capacity to allow some burstyness. We should probably also explain in the cost version that elements costing more than the allowed burst will be delayed proportionally to their cost minus available tokens, meeting the target rate.